x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/RBACEngine.java

```diff
@@ -132,6 +135,50 @@ private void getRoles(User user, Authentication authentication, ActionListener<R
         rolesStore.getRoles(user, authentication, listener);
     }
 
+    @Override
+    public boolean isChildActionAuthorizedByParent(RequestInfo requestInfo, String parentAction, IndicesAccessControl parentAccessControl) {
+        final String childAction = requestInfo.getAction();
+        if (IndexPrivilege.CREATE_INDEX_MATCHER.test(childAction) || childAction.equals(TransportShardBulkAction.ACTION_NAME)) {
+            // These need special handling, so don't short circuit
+            return false;
+        }
+        if (isScrollRelatedAction(childAction) || isAsyncRelatedAction(childAction)) {
+            // These need special handling, so don't short circuit
+            return false;
+        }
```

Comment:
I think this check is redundant since the logic already mandates the request must be an instance of `IndicesRequest`.